Src/WitsmlExplorer.Api/Repositories/CosmosRepository.cs

```diff
@@ -55,26 +55,23 @@ await foreach (var document in container.GetItemQueryIterator<TDocument>(queryDe
         public async Task<TDocument> UpdateDocumentAsync(TDocumentId id, TDocument document)
         {
             var container = cosmosClient.GetContainer(dbName, containerId);
-            var itemResponse = await container.ReplaceItemAsync(document, document.Id.ToString());
-            return itemResponse.Value;
+            return await container.ReplaceItemAsync<TDocument>(document, document.Id.ToString());
         }
 
         public async Task<TDocument> CreateDocumentAsync(TDocument document)
         {
             var container = cosmosClient.GetContainer(dbName, containerId);
-            var createResponse = await container.CreateItemAsync(document);
-
-            return createResponse.Value;
+            return await container.CreateItemAsync<TDocument>(document, new PartitionKey(document.Id.ToString()));
         }
 
         public async Task DeleteDocumentAsync(TDocumentId id)
         {
             var container = cosmosClient.GetContainer(dbName, containerId);
 
             var deleteResponse = await container.DeleteItemStreamAsync(id.ToString(), new PartitionKey(id.ToString()));
-            if (deleteResponse.Status != 204)
+            if (deleteResponse.StatusCode != HttpStatusCode.NoContent)
             {
-                throw new RepositoryException($"Unable to delete document with id: {id}", deleteResponse.Status);
+                throw new RepositoryException($"Unable to delete document with id: {id}", (int) deleteResponse.StatusCode);
```

Comment:
The comment suggests that it is the document id that is printed (`Unable to delete document with id: `), but it is the statusCode that is placed here. Seems to be an old issue, but I suggest replacing `deleteResponse.StatusCode` with `id` here